File: backend/app/schemas/integrity.py
from pydantic import BaseModel


class IntegrityCheckCreate(BaseModel):
    repair_thumbnails: bool = True


class IntegrityIssue(BaseModel):
    code: str
    message: str
    path: str | None = None
    entity_type: str
    entity_id: int
    repairable: bool
    repaired: bool = False


class IntegrityReport(BaseModel):
    project_id: int
    checked_videos: int
    checked_frames: int
    issue_count: int
    repaired_count: int
    issues: list[IntegrityIssue]


class IntegrityReportEnvelope(BaseModel):
    data: IntegrityReport
    error: None = None

File: frontend/src/services/integrity.ts
import { apiRequest } from "./api";

export interface IntegrityIssue {
  code: string;
  message: string;
  path: string | null;
  entity_type: string;
  entity_id: number;
  repairable: boolean;
  repaired: boolean;
}
export interface IntegrityReport {
  project_id: number;
  checked_videos: number;
  checked_frames: number;
  issue_count: number;
  repaired_count: number;
  issues: IntegrityIssue[];
}
export const runIntegrityCheck = async (
  projectId: number,
  repairThumbnails: boolean,
) =>
  (
    await apiRequest<{ data: IntegrityReport }>(
      `/projects/${projectId}/integrity-check`,
      {
        method: "POST",
        body: JSON.stringify({ repair_thumbnails: repairThumbnails }),
      },
    )
  ).data;
